src/twisted/internet/tcp.py

```diff
@@ -100,6 +100,39 @@ class _TLSServerMixin(object):
 _portNameType = (str, unicode)
 
 
+def _getrealname(addr):
+    """
+    Return a 2-tuple of socket IP and port for IPv4 and a 4-tuple of
+    socket IP, port, flowInfo, and scopeID.  With IPv6 addresses, this
+    preserves the interface portion.
```

Comment:
Out of "socket IP, port, flowInfo, and scopeID", none of those things are obviously called "interface".  Can you clarify just a little as to what is being preserved, and why this is important (or what wouldn't preserve it?)  I think for future maintainers understanding that this is papering over a 3.6 / 3.7 difference is probably pretty important.